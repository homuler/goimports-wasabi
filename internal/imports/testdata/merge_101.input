// merge comments
package main

// doc comment (1)
import /* prelparen comment (1) */ ( // postlparen comment

	// stdlib doc comment (1)

	/* path comment for fmt (1) */ "fmt" // comment for fmt (1)
	// trailing comment for fmt (1)

	// foreign doc comment (1)

	// doc comment for foo
	"github.com/homuler/foo" // comment for foo
	// trailing comment for foo

	// bottom comment (1)
) // postrparen comment (1)
// trailing comment

// header comment (1)

// doc comment for context
import "context" // comment for context
// trailing comment for context

// doc comment for fmt
import /* path comment for fmt (2) */ "fmt" /* comment for fmt (2)
*/ // comment for fmt (3)
/* trailing comment for fmt (2)
*/

// header comment (2)

// doc comment for baz
import /* path comment for baz */ "local/baz" // comment for baz
// doc comment (2)
import /* prelparen comment (2) */ (
	// local doc comment

	"local/qux"

	// stdlib doc comment (2)

	/* name comment for f */ f /* path comment for fmt */ "fmt" // comment for fmt

	// foreign doc comment (2)

	// doc comment for bar
	"github.com/homuler/bar" /* comment
	for bar */ // comment for bar
	// trailing comment for bar

	// bottom comment (2)
/* bottom comment (3) */) // postrparen comment (2)
// comment for var
var (
	x = foo.X
	y = bar.Y
	w = baz.W
	z = qux.Z
)

func main() {
	fmt.Println(context.Background())
	f.Printf("%v %v %v %v\n", x, y, w, z)
}
