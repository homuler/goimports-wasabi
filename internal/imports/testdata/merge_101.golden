// merge comments
package main

// header comment (1)

// header comment (2)

// doc comment (1)
// doc comment (2)
import /* prelparen comment (1) */ /* prelparen comment (2) */ ( // postlparen comment

	// stdlib doc comment (1)

	// stdlib doc comment (2)

	// doc comment for context
	"context" // comment for context
	// trailing comment for context

	// doc comment for fmt
	/* path comment for fmt (1) */ /* path comment for fmt (2) */
	"fmt"                          // comment for fmt (1)
	/* comment for fmt (2)
	 */ // comment for fmt (3)
	// trailing comment for fmt (1)
	/* trailing comment for fmt (2)
	 */

	/* name comment for f */
	f /* path comment for fmt */ "fmt" // comment for fmt

	// foreign doc comment (1)

	// foreign doc comment (2)

	// doc comment for bar
	"github.com/homuler/bar" /* comment
	for bar */ // comment for bar
	// trailing comment for bar

	// doc comment for foo
	"github.com/homuler/foo" // comment for foo
	// trailing comment for foo

	// local doc comment

	// doc comment for baz
	/* path comment for baz */
	"local/baz" // comment for baz
	"local/qux"
	// bottom comment (1)
	// bottom comment (2)
	/* bottom comment (3) */) // postrparen comment (1)
// postrparen comment (2)
// trailing comment

// comment for var
var (
	x = foo.X
	y = bar.Y
	w = baz.W
	z = qux.Z
)

func main() {
	fmt.Println(context.Background())
	f.Printf("%v %v %v %v\n", x, y, w, z)
}
